usr/share/rear/rescue/GNU/Linux/290_kernel_cmdline.sh

```diff
@@ -1,51 +1,60 @@
+# Purpose of the script is to get the COPY_KERNEL_PARAMETERS on the current system
+# to be used in the rescue/recovery system via automated update of KERNEL_CMDLINE.
 
-# Scanning current kernel cmdline to look for important option ($COPY_KERNEL_PARAMETERS) to include in KERNEL_CMDLINE
-for current_kernel_option in $( cat /proc/cmdline ); do
+# Also protect the rescue/recovery system by removing net.ifnames=0 from KERNEL_CMDLINE
+# if net.ifnames=0 is in KERNEL_CMDLINE but persistent network interface naming is used:
+local persistent_naming='no'
+is_persistent_ethernet_name $( ip route | awk '$2 == "dev" && $8 == "src" { print $3 }' | sort -u | head -n1 ) && persistent_naming='yes'
+
+# Scan current kernel cmdline for options in COPY_KERNEL_PARAMETERS to be included in KERNEL_CMDLINE:
+local current_kernel_option
+local new_kernel_options_to_add=()
+for current_kernel_option in $( cat /proc/cmdline ) ; do
     # Get the current kernel option name (part before leftmost "=") and
     # add the whole option (with value) to new_kernel_options_to_add array
     # if the option name is part of COPY_KERNEL_PARAMETERS array:
-    if IsInArray "${current_kernel_option%%=*}" "${COPY_KERNEL_PARAMETERS[@]}" ; then
-        new_kernel_options_to_add+=( "$current_kernel_option" )
-    fi
+    IsInArray "${current_kernel_option%%=*}" "${COPY_KERNEL_PARAMETERS[@]}" && new_kernel_options_to_add+=( "$current_kernel_option" )
 done
 
-# Verify if the kernel option we want to add to KERNEL_CMDLINE are not already set/force by the user in the rear configuration.
-# If yes, the parameter set in the configuration file have the priority and superseed the current kernel option.
+# Check if the kernel options we want to add to KERNEL_CMDLINE are already set by the user in KERNEL_CMDLINE.
+# If yes, the user setting has priority and superseds the kernel option from the current system.
+# For the check use the existing KERNEL_CMDLINE when this script is started
+# and not the modified KERNEL_CMDLINE with already added kernel options
+# to make it possible to add several kernel options by this script
+# with same kernel option keyword like console=ttyS0,9600 console=tty0
+# see https://github.com/rear/rear/pull/2749#issuecomment-1197843273
+# and https://github.com/rear/rear/pull/2844
+local existing_kernel_cmdline="$KERNEL_CMDLINE"
+local existing_kernel_option new_kernel_option new_kernel_option_keyword
 for new_kernel_option in "${new_kernel_options_to_add[@]}" ; do
     new_kernel_option_keyword="${new_kernel_option%%=*}"
-
-    for rear_kernel_option in $KERNEL_CMDLINE ; do
-        # Check if a kernel option key without value parameter (everything before =) is not already present in rear KERNEL_CMDLINE array.
-        if test "$new_kernel_option_keyword" = "${rear_kernel_option%%=*}" ; then
-            LogPrintError "Not adding '$new_kernel_option' (superseded by existing '$rear_kernel_option' in KERNEL_CMDLINE)"
```

Comment:
I like the replacement of `LogPrintError` by `LogPrint`, it is consistent with the "warnings are the waste of time" spirit.

Reply:
My LogPrintError was wrong because nothing is wrong here
so LogPrint is used to show a normal information message.

The question is if LogPrint is OK for those messages
in this script or if it is perhaps oververbose?
If it is oververbose DebugPrint would be right.

Regarding "WARNING is a waste of my time"
https://schlomo.schapiro.org/2015/04/warning-is-waste-of-my-time.html
therein is a link "discussion about the use of the WARNING" to
https://github.com/rear/rear/issues/564
therein see those three subsequent issue comments
https://github.com/rear/rear/issues/564#issuecomment-86418677
https://github.com/rear/rear/issues/564#issuecomment-86441426
https://github.com/rear/rear/issues/564#issuecomment-86458179
in particular in the latter
```
The issue is not that a program must run
either silently successfully
or abort with an ERROR message,
the issue is only that WARNING messages are useless.
```
The crucial point is the word "WARNING" that is useless
(or a log level for info that is attributed to be a warning).
For example in
```
LogPrint "WARNING something is not as it is expected to be"
LogPrintError "WARNING something is not as it is expected to be"
DebugPrint "WARNING something is not as it is expected to be"
```
the word "WARNING" is useless so it should better be
```
LogPrint "Something is not as it is expected to be"
LogPrintError "Something is not as it is expected to be"
DebugPrint "Something is not as it is expected to be"
```
i.e. only the plain information without a prejudgement
attribute what severity it means for the user.
Perhaps the user already knows that
"something is not as it is expected to be"
because he has intentionally set up 'something' this way.
Perhaps when "something is not as it is expected to be"
it is catastrophic for the user (i.e. "CRITICAL ERROR").